dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -728,18 +728,23 @@ public boolean equals(Object obj) {
             if (obj == null || !(obj instanceof LocalVariable)) {
                 return false;
             }
+
             LocalVariable other = (LocalVariable) obj;
             VarSymbol vs = (VarSymbol) element;
             VarSymbol vsother = (VarSymbol) other.element;
+            // The code below isn't just return vs.equals(vsother) because an element might be different
+            // between subcheckers.
+
             // Use TypeAnnotationUtils.unannotatedType(type).toString().equals(...) instead of
             // Types.isSameType(...) because Types requires a processing environment, and
             // FlowExpressions is designed to be independent of processing environment.  See also
             // calls to getType().toString() in FlowExpressions.
             return vsother.name.contentEquals(vs.name)
+                    && vsother.owner.toString().equals(vs.owner.toString())
+                    && vs.pos == vsother.pos
```

Comment:
I would move the `pos` comparison first, as it is the cheapest one to perform and most likely to fail.
Do we still need the type comparison? Can two symbols have the same position, name, and owner but have different types?
An update to the comment above this condition would also help in understanding why we need to consider the position.
Symbols from bytecode have no `pos`, I assume; I guess this isn't ever a problem for this.